src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,600 @@
+  KeyboardEventHandler,
+  MouseEventHandler,
+  PointerEventHandler,
+  ReactElement,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { GridOnScrollProps, VariableSizeGrid } from "react-window";
+import AutoSizer, { Size } from "react-virtualized-auto-sizer";
+import RowNumberColumn from "./RowNumberColumn";
+import Header from "./Header";
+import styled from "styled-components";
+import { GridContextMenuItemProps, GridProps, ItemDataType, onSelectFn } from "./types";
+import { useSelectionActions } from "./useSelectionActions";
+import { useRefCallback } from "./useRefCallback";
+import { mergeRefs } from "@/utils/mergeRefs";
+import { Cell } from "./Cell";
+import { ContextMenu } from "@/components";
+
+const NO_BUTTONS_PRESSED = 0;
+const RIGHT_BUTTON_PRESSED = 2;
+
+const GridContainer = styled.div`
+  display: flex;
+  flex-direction: column-reverse;
+  background: ${({ theme }) => theme.click.grid.body.cell.color.background.default};
+  user-select: none;
+  overflow-anchor: none;
+`;
+
+const getRenderedCursor = (children: Array<ReactElement>) =>
+  children.reduce(
+    ([minRow, maxRow, minColumn, maxColumn], { props: { columnIndex, rowIndex } }) => {
+      if (rowIndex < minRow) {
+        minRow = rowIndex;
+      }
+      if (rowIndex > maxRow) {
+        maxRow = rowIndex;
+      }
+      if (columnIndex < minColumn) {
+        minColumn = columnIndex;
+      }
+      if (columnIndex > maxColumn) {
+        maxColumn = columnIndex;
+      }
+
+      return [minRow, maxRow, minColumn, maxColumn];
+    },
+    [
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+      Number.POSITIVE_INFINITY,
+      Number.NEGATIVE_INFINITY,
+    ]
+  );
+
+const GridDataContainer = styled.div<{ $top: number; $left: number }>`
+  position: absolute;
+  top: 0;
+  left: 0;
+  ${({ $top, $left }) => `
+    margin-top: ${$top}px;
+    margin-left: ${$left}px;
+  `}
+`;
+
+const ContextMenuTrigger = styled.div`
+  outline: none;
+  height: 100%;
+  width: 100%;
+`;
+
+interface InnerElementTypeTypes extends HTMLAttributes<HTMLDivElement> {
+  children: Array<ReactElement>;
+}
+
+const OuterElementType = forwardRef<HTMLDivElement>((props, ref) => (
+  <div
+    ref={ref}
+    data-testid="grid-outer-element"
+    {...props}
+  />
+));
+
+export const Grid = forwardRef<VariableSizeGrid, GridProps>(
+  (
+    {
+      rowStart = 0,
+      showRowNumber = true,
+      rounded = "none",
+      showHeader = true,
+      focus,
+      useIsScrolling = true,
+      rowHeight = 33,
+      columnCount,
+      columnWidth,
+      onSelect: onSelectProp,
+      headerHeight = 33,
+      rowCount,
+      cell,
+      onColumnResize: onColumnResizeProp,
+      onFocusChange: onFocusChangeProp,
+      getMenuOptions,
+      onKeyDown: onKeyDownProp,
+      ...props
+    },
+    forwardedRed
+  ) => {
+    const [menuOptions, setMenuOptions] = useState<Array<GridContextMenuItemProps>>([]);
```

Comment:
we should have a default menu options.
when using the grid i wouldn't want to pass menu options every time.

I don't know what useful options could be though, maybe "copy as string", "copy as json". wdyt? cc @fneves @ArctypeZach

Reply:
@serdec I didn't add the copy in as a default option because of the following reason

- Currently, the Cell is a component. We can make it to have the contents as a string using a function. The copy needs the cell values and thats why i made it exterior and created a function that could process the functionality
- The second reason is that the copy might not be needed everywhere and we might only need to show the data like maybe in the Data loading etc
- Also the inspect cell and json data are not added in click-ui for the same reason

Reply:
I would say we should have defaults. By default it should copy to the clipboard, if you pass in a function it uses that function.
We can have a prop that disables the whole menu options for the dataloading use case.

Reply:
@fneves @serdec Just a doubt for the default copy functionality how should we implement the createToast for success and failure
Should we add it using a prop in Grid for copy success and failure?

Reply:
as a user i wouldn't expect the copy function to fail as i wouldn't expect the `console.log` to fail unless there is some critical problem with my js. If any error happens we should log it into the console